go/adbc/driver/snowflake/record_reader.go

```diff
@@ -567,6 +567,17 @@ func newRecordReader(ctx context.Context, alloc memory.Allocator, ld gosnowflake
 
 	lastChannelIndex := len(chs) - 1
 	go func() {
+		// place this here so that we always clean up, but they can't be in a
+		// separate goroutine. Otherwise we'll have a race condition between
+		// the call to wait and the calls to group.Go to kick off the jobs
+		// to perform the pre-fetching (GH-1283).
```

Comment:
Should this be placed after the loop below then? That way we absolutely know group.Go has been called

Reply:
We always call `group.Go` once above this goroutine in order to kick off the first batch. In addition, if `group.Go` has never been called, `group.Wait` will just return immediately.

Hmm, though re-reading the code. We probably could just place this after the for loop and we don't need the defer. I'll update this